java/memory/src/test/java/org/apache/arrow/memory/TestBaseAllocator.java

```diff
@@ -478,6 +520,79 @@ public void testAllocator_overAllocateParent() throws Exception {
     }
   }
 
+  @Test
+  public void testAllocator_parentLimit_outcomeDetails() throws Exception {
+    try (final RootAllocator rootAllocator =
+        new RootAllocator(MAX_ALLOCATION)) {
+      try (final BufferAllocator childAllocator =
+          rootAllocator.newChildAllocator("child", 0, MAX_ALLOCATION / 2)) {
+        try (final BufferAllocator grandChildAllocator =
+            childAllocator.newChildAllocator("grandchild", MAX_ALLOCATION / 4, MAX_ALLOCATION)) {
+          try {
+            final ArrowBuf arrowBuf = grandChildAllocator.buffer(MAX_ALLOCATION);
+            fail("allocated memory beyond max allowed");
+          } catch (OutOfMemoryException e) {
+
+            // expected
+            AllocationOutcomeDetails outcomeDetails = e.getOutcomeDetails();
+            assertNotNull("missing outcome details", outcomeDetails);
+
+            assertEquals(outcomeDetails.getFailedAllocator(), childAllocator);
+
```

Comment:
maybe add a comment indicating the order should be in the same order the allocators where created in?

Reply:
added a comment - allocators should be in the order from child till root